Predictions with weighted models should match prod

charleslparker opened the pull request:
/cc @akashenfelter @ashenfad 

This makes the predictions with weighted models match what we have in wintermute right now.  Two questions:

1. I have a weak opinion that the current behavior in `Tree` with regard to the returned distribution is a bug, which is why I changed it.  My primary reasoning is that the `predict_proportional` method returns  an aggregate of the weighted distribution for weighted models, so it seemed non-proportional predictions should do the same.  In any case, we just have to make sure that the weighted distribution makes it to the caller in one way or another and I'm fairly agnostic as to how that happens.

2. I haven't added a test for this yet.  The test framework we're using for this project is somewhat opaque to me and I think I did it wrong last time.  If someone could let me know where to add this, that would be excellent.  A sufficient test looks like this:

```
from bigml.api import BigML
from bigml.ensemble import Ensemble

api = BigML(storage='./storage')
resource = api.create_source('./data/iris.csv')

dataset = api.create_dataset(resource)
api.ok(dataset)

ensemble = api.create_ensemble(dataset, {"objective_weights":
                                         [["Iris-versicolor", 1],
                                          ["Iris-virginica", 2],
                                          ["Iris-setosa", 4]],
                                         "ensemble_sample": {'seed': 'test'}})
api.ok(ensemble)

flower = {"petal width": 0.5, "sepal width": 1.0}

remote = api.create_prediction(ensemble, flower)['object']['probabilities']
remote = {p[0]: p[1] for p in remote}

local_ensemble = Ensemble(ensemble)
local = local_ensemble.predict_probability(flower)
local = {p['prediction']: p['probability'] for p in local}

classes = local.keys()

for rp, lp in zip([remote[c] for c in classes], [local[c] for c in classes]):
    assert abs(rp - lp) < 1e-5, str((rp, lp))
```

mmerce commented on bigml/model.py:
```diff
@@ -373,11 +373,16 @@ def predict_probability(self, input_data, by_name=True,
-            total = float(sum([category[1] for category in root_dist]))
-            category_map = {category[0]: category[1] / total
+
+            if not self.tree.weighted:
```
you can ignore this nit, but I would reverse the condition to avoid an additional not

```python
if self.tree.weighted:
    ...
else:
  ...
```

charleslparker commented:
@mmerce - Thanks for the explanation; I think I understand a bit better now.  I added some tests and fixed the nit so this should be good.

@akashenfelter - This should be good for you to start integrating into the PHP PR, unless Merce sees any major problems at the last minute.

mmerce commented:
@charleslparker - I just added an entry to the HISTORY.rst file where we log the changes in this version, so please rebase to the bigmlcom/charleslparker-master branch. I'll close this PR and open another one from that branch, because the integrated tests need to be run from a branc in bigmlcom for the environment variables to be properly set.
